cassis/cas.py

```diff
@@ -612,33 +614,52 @@ def _find_all_fs(self, generate_missing_ids: bool = True) -> Iterable[FeatureStr
             all_fs[fs.xmiID] = fs
 
             t = ts.get_type(fs.type)
+
+            # Arrays contents are handled separately - they only have one "virtual" feature: elements
+            if t.supertypeName == "uima.cas.ArrayBase":
+                if t.name == "uima.cas.FSArray":
+                    for ref in fs.elements:
+                        if ref.xmiID in all_fs:
+                            continue
+                        openlist.append(ref)
+                continue
+
+            # For non-array types, we look at the features - this includes also FSList-types
             for feature in t.all_features:
                 feature_name = feature.name
 
                 if feature_name == "sofa":
                     continue
 
+                if ts.is_primitive(feature.rangeTypeName):
+                    continue
+
+                feature_value = getattr(fs, feature_name)
+                if feature_value is None:
+                    continue
+
                 if (
-                    ts.is_primitive(feature.rangeTypeName)
-                    or ts.is_primitive_collection(feature.rangeTypeName)
-                    or ts.is_primitive_collection(fs.type)
+                    include_inlinable_arrays is False
+                    and feature.multipleReferencesAllowed is not True
```

Comment:
Why dont you write `not feature.multipleReferencesAllowed`? Similar question to several other locations

Reply:
Because `feature.multipleReferencesAllowed` can be either `False` or `None` or `True` - if it is not `True`, then the condition should kick in. Not sure if it is sensible to write this as `not feature.multipleReferencesAllowed` - I would expect that `None` is not negatable.

Reply:
```
>>> bool(None)
False
>>> bool(not None)
True

```